### PR TITLE
[Merged by Bors] - feat(Combinatorics/SimpleGraph): bot is not connected

### PR DESCRIPTION
**Pull request by Rida-Hamadani**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by YaelDillies — COMMENTED** [review]
Can we also have `bot_not_preconnected`?

**Comment by YaelDillies** [comment]
Thanks!

maintainer merge

File: Mathlib/Combinatorics/SimpleGraph/Path.lean [hunk 1]
```diff
@@ -719,6 +719,14 @@ theorem Preconnected.map {G : SimpleGraph V} {H : SimpleGraph V'} (f : G →g H)
 protected lemma Preconnected.mono  {G G' : SimpleGraph V} (h : G ≤ G') (hG : G.Preconnected) :
     G'.Preconnected := fun u v => (hG u v).mono h
 
+lemma bot_preconnected_iff_subsingleton : (⊥ : SimpleGraph V).Preconnected ↔ Subsingleton V := by
+  refine ⟨fun h ↦ ?_, fun h ↦ by simpa [subsingleton_iff, ← reachable_bot] using h⟩
+  contrapose h
+  simp [nontrivial_iff.mp <| not_subsingleton_iff_nontrivial.mp h, Preconnected, reachable_bot, h]
+
+lemma bot_not_preconnected [Nontrivial V] : ¬(⊥ : SimpleGraph V).Preconnected :=
```
Comment (b-mehta) [inline Mathlib/Combinatorics/SimpleGraph/Path.lean:727]: Let's also add that it's connected under the typeclass assumption of subsingleton

**Review by b-mehta — APPROVED** [review]
bors d+

File: Mathlib/Combinatorics/SimpleGraph/Path.lean (hunk 1 above)
Comment (Rida-Hamadani) [reply Mathlib/Combinatorics/SimpleGraph/Path.lean:727]: We will need to assume non-emptiness too, right?
We can however do preconnectedness without more assumptions. Is that enough?

Comment (b-mehta) [reply Mathlib/Combinatorics/SimpleGraph/Path.lean:727]: Yup, go ahead

**Pull request closed without merging**